cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedSparkProgramRunner.java

```diff
@@ -76,9 +77,15 @@ protected ProgramController launch(Program program, ProgramOptions options,
     SparkSpecification spec = appSpec.getSpark().get(program.getName());
     Preconditions.checkNotNull(spec, "Missing SparkSpecification for %s", program.getId());
 
-    // Localize the spark-assembly jar
+    // Localize the spark-assembly jar and spark conf zip
     File sparkAssemblyJar = SparkUtils.locateSparkAssemblyJar();
-    localizeResources.put(sparkAssemblyJar.getName(), new LocalizeResource(sparkAssemblyJar));
+    try {
+      sparkAssemblyJar = SparkUtils.getRewrittenSparkAssemblyJar(cConf);
+      localizeResources.put(sparkAssemblyJar.getName(), new LocalizeResource(sparkAssemblyJar));
+    } catch (IOException e) {
+      LOG.warn("Failed to located rewritten Spark Assembly JAR. Fallback to use the original jar.", e);
```

Comment:
"Failed to locate the rewritten Spark Assembly JAR. Falling back to use the original jar."